### ISSUE TITLE
Streamlined install

### ISSUE THREAD

**Issue by KrzysztofMadejski** [edgi-govdata-archiving/web-monitoring-ui#72]
While installing web-monitoring-ui I've run into several issues. I propose following changes. Let me know what you think about them. I'll send PR when you accept them
- change [.env.example](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/.env.example) to default DB development config:
  ```
  WEB_MONITORING_DB_URL=http://localhost:3000

  # These will be removed once a login form is added
  WEB_MONITORING_DB_USER=seed-admin@example.com
  WEB_MONITORING_DB_PASSWORD=PASSWORD
  ```
- run gulp by `npm run gulp` to use local version in node_modules/.bin
- (optional) add Troubleshooting section providing hints `The program 'gulp' is currently not installed`
- create gulp's `build` supertask that won't hang on `watch` as the default task is doing (I was waiting for sth to happen for quite a time)
- change default PORT to 3001, because web-monitoring-db runs on 3000
- drop `yarn`
  First time I hear about it. Why not use quite popular npm? What yarn offers more? I've managed to setup the app using `npm install` and `npm run start`

**Comment by lightandluck** [comment]
> change .env.example

I think this should be left pointing at the staging site, so contributors to ui don't have to get db setup and running, less friction to get working on just this project.

> run `npm run gulp`

+1, makes sense to have gulp in devDependencies and run locally so everyone is working with the same version.

> add Troubleshooting section

+1, more documentation is great!

> create `build` task 

If this helps alleviate some confusion +1. We still want to keep the `watch` task to make development easier. If you want to resolve #71 at the same time, that would be appreciated! But not necessary.

> change PORT to 3001

+1, I haven't had to run both projects (db & ui) at the same time, but if this solves that problem, go for it. Not married to port numbers.

> drop `yarn`

@Mr0grog should speak to that

**Comment by Mr0grog** [comment]
> change .env.example to default DB development config:

I’m not sure that you should necessarily always be developing against your local environment. I set it to staging so that someone can pull down UI and immediately start working on it without setting up any other other services.

> run gulp by `npm run gulp` to use local version in `node_modules/.bin`

One alternative to this is to use Gulp’s recommended method: globally install `gulp-cli`, which gives you a `gulp` command that searches for a locally installed Gulp (e.g. the one in the current package’s `node_modules`) and uses that.

> create gulp's build supertask that won't hang on watch as the default task is doing (I was waiting for sth to happen for quite a time)

I thought there was an issue for this, but I can’t find it. I’ve noted it before in some comments at least, so 👍 👍 👍 

> drop yarn… Why not use quite popular npm? What yarn offers more?

Ah! Yarn has a *a lot* of advantages over NPM (it was specifically created to solve many of NPM’s shortcomings). Some things it does:

- It is *much* faster
- Dependencies take up dramatically less space on disk (they are all symlinked from a central place)
- Its default mode is to include only a *single* version of a given package that satisfies all requirements (in NPM, if you have three dependencies that each depend on different versions of a fourth dependency, you’ll wind up with 3 versions of that last dependency. More often than not, Yarn successfully makes sure you only have *one* of that dependency, which is especially useful when building browser bundles)
- Its lockfiles (and how it uses them) solve the problems in NPM shrinkwrap files much better (consistency issues, how dependencies are unwound, etc. — lots of blog posts on this)
- It generally doesn’t allow to install a package without also ensuring it gets added to `package.json` and `yarn.lock`, which helps alleviate a lot of simple mistakes

There is a lot written out on the internet about these issues. NPM 5 improves a lot—it is close to Yarn in speed, but not quite there, for example—but doesn’t quite catch up in all things. It’s possible the differences will eventually become negligible and projects will merge! But I think, for now, Yarn is a big improvement.

**Comment by Mr0grog** [comment]
Other things I didn’t mention sound like good ideas to me—should have stated that explicitly.

**Comment by KrzysztofMadejski** [comment]
Closed in #80
